### PR TITLE
Add type support to PropertyConfigurator

### PR DESCRIPTION
**Pull request by guperrot**
Code + UI fix (no change requiring a screenshot compared to #1164 screenshot).

### CODE REVIEW

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m [hunk 1]
```diff
@@ -374,122 +378,117 @@ - (void)testLongListOfSubChildren {
...
+  [expectedProperties setBool:propTrackValue2 forKey:propTrackKey2];
+  MSEventProperties *trackEventProperties = [MSEventProperties new];
+  [trackEventProperties setString:propTrackValue forKey:propTrackKey];
+  [trackEventProperties setBool:propTrackValue2 forKey:propTrackKey2];
+  
   // When
-  [target trackEvent:eventName withProperties:@{propTrackKey : propTrackValue, propTrackKey2 : propTrackValue2}];
-
+  [target trackEvent:eventName withTypedProperties:trackEventProperties];
+
```
Comment (jaelim-ms) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m]: Run formatter.

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSPropertyConfiguratorTests.m [hunk 2]
```diff
@@ -6,51 +6,38 @@
 #import "MSPropertyConfiguratorInternal.h"
 #import "MSPropertyConfiguratorPrivate.h"
 #import "MSTestFrameworks.h"
+#import "MSStringTypedProperty.h"
+#import "MSDateTimeTypedProperty.h"
```
Comment (jaelim-ms) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSPropertyConfiguratorTests.m]: Sort import.

File: AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h [hunk 3]
```diff
@@ -37,6 +37,54 @@ NS_ASSUME_NONNULL_BEGIN
...
 - (void)setEventPropertyString:(NSString *)propertyValue forKey:(NSString *)propertyKey;
 
+/**
+ * Set a double event property to be attached to events tracked by this transmission target and its child transmission targets.
+ *
+ * @param propertyValue Property value. Must be finite (`NAN` and `INFINITY` not allowed).
+ * @param propertyKey Property key.
+ *
+ * @discussion A property set in a child transmission target overrides a property with the same key inherited from its parents. Also, the
+ * properties passed to the `trackEvent:withProperties:` override any property with the same key from the transmission target itself or its
```
Comment (jwallra) [inline AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h]: To be technically correct, this should be "`trackEvent:withProperties:` *or* `trackEvent:withTypedProperties:` overrides", yes?

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m [hunk 4]
```diff
@@ -374,120 +378,115 @@ - (void)testLongListOfSubChildren {
   }
 }
 
-- (void)testSetAndRemoveEventProperty {
```
Comment (jwallra) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m]: How is this case being covered now?  Why was this test removed?

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSPropertyConfiguratorTests.m [hunk 5]
```diff
@@ -103,4 +89,102 @@ - (void)testDeviceIdDoesNotPropagate {
...
+  // When
+  [self.sut removeEventPropertyForKey:@"APropKey"];
+
+  // Then
+  XCTAssertTrue([self.sut.eventProperties isEmpty]);
+}
+
+- (void)testSetAndRemoveEventPropertiesWithNilKeys {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnonnull"
```
Comment (jwallra) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSPropertyConfiguratorTests.m]: Let's clean up the `#pragma`s in this file... New lines before comments still applies.  It looks like we're inconsistent but I would like to see new lines before and after groups of `#pragma`s as well

Comment (jwallra) [reply AppCenterAnalytics/AppCenterAnalyticsTests/MSPropertyConfiguratorTests.m]: Either that or don't put them before comments; just put them directly adjacent to the code they're configuring.,

File: AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h (hunk 3 above)
Comment (clpolet) [reply AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h]: Yes right, updating it to add `trackEvent:withTypedProperties:`

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m (hunk 4 above)
Comment (clpolet) [reply AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m]: It's been divided into smaller tests for more readability :)

**Pull request merged**